File: gateway/src/main/java/com/beat/gateway/EnableGatewayConfig.java
package com.beat.gateway;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.springframework.context.annotation.Import;

import com.beat.gateway.shared.internal.GatewayConfigImportSelector;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Import(GatewayConfigImportSelector.class)
public @interface EnableGatewayConfig {

	GatewayConfigGroup[] value();
}

File: gateway/src/main/kotlin/com/beat/gateway/EnableGatewayConfig.kt
package com.beat.gateway

import com.beat.gateway.shared.internal.GatewayConfigImportSelector
import org.springframework.context.annotation.Import

@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@Import(GatewayConfigImportSelector::class)
annotation class EnableGatewayConfig(
    val value: Array<GatewayConfigGroup>,
)
